homeassistant/components/airtouch5/climate.py

```diff
@@ -0,0 +1,388 @@
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Add data updated listener after this object has been initialized."""
+        await super().async_added_to_hass()
+        self._client.ac_status_callbacks.append(self._async_update_attrs)
+        self._async_update_attrs(self._client.latest_ac_status)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Remove data updated listener after this object has been initialized."""
+        await super().async_will_remove_from_hass()
+        self._client.ac_status_callbacks.remove(self._async_update_attrs)
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new operation mode."""
+        set_power_setting: SetPowerSetting
+        set_ac_mode: SetAcMode
+
+        if hvac_mode == HVACMode.OFF:
+            set_power_setting = SetPowerSetting.SET_TO_OFF
+            set_ac_mode = SetAcMode.KEEP_AC_MODE
+        else:
+            set_power_setting = SetPowerSetting.SET_TO_ON
+            if hvac_mode not in HVAC_MODE_TO_SET_AC_MODE:
+                raise ValueError(f"Unsupported hvac mode: {hvac_mode}")
+            set_ac_mode = HVAC_MODE_TO_SET_AC_MODE[hvac_mode]
+
+        control = AcControl(
+            set_power_setting,
+            self._ability.ac_number,
+            set_ac_mode,
+            SetAcFanSpeed.KEEP_AC_FAN_SPEED,
+            SetpointControl.KEEP_SETPOINT_VALUE,
+            0,
+        )
+        packet = self._client.data_packet_factory.ac_control([control])
+        await self._client.send_packet(packet)
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set new fan mode."""
+        if fan_mode not in FAN_MODE_TO_SET_AC_FAN_SPEED:
+            raise ValueError(f"Unsupported fan mode: {fan_mode}")
+        fan_speed = FAN_MODE_TO_SET_AC_FAN_SPEED[fan_mode]
+        control = AcControl(
+            SetPowerSetting.KEEP_POWER_SETTING,
+            self._ability.ac_number,
+            SetAcMode.KEEP_AC_MODE,
+            fan_speed,
+            SetpointControl.KEEP_SETPOINT_VALUE,
+            0,
+        )
+        packet = self._client.data_packet_factory.ac_control([control])
+        await self._client.send_packet(packet)
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperatures."""
+        if (temp := kwargs.get(ATTR_TEMPERATURE)) is None:
+            _LOGGER.debug("Argument `temperature` is missing in set_temperature")
+            return
+
+        control = AcControl(
+            SetPowerSetting.KEEP_POWER_SETTING,
+            self._ability.ac_number,
+            SetAcMode.KEEP_AC_MODE,
+            SetAcFanSpeed.KEEP_AC_FAN_SPEED,
+            SetpointControl.CHANGE_SETPOINT,
+            float(temp),
+        )
+        packet = self._client.data_packet_factory.ac_control([control])
+        await self._client.send_packet(packet)
```

Comment:
Please move this to a separate function, which is then called with different parameters like 

```python
async def _control(*, power: SetPowerSetting = SetPowerSetting.KEEP_POWER_SETTING, mode: SetAcMode = SetAcMode.KEEP_AC_MODE, ....
```
So you need only to specify the params, which need to be different than the default vaule

Reply:
This isn't going to save any code, we create 3 AcControl objects and only 2 of them are similar-ish.

Reply:
```suggestion
        self._control(temp=temp)
```

This saves duplicate code and improves readability and maintainability.